unit.go

```diff
@@ -296,7 +355,7 @@ func (u *unit) Validate() error {
 	if u.WorkloadStatus_ == nil {
 		return errors.NotValidf("unit %q missing workload status", u.Name_)
 	}
-	if u.Tools_ == nil {
+	if u.Tools_ == nil && u.Type_ != "caas" {
```

Comment:
After seeing "caas" a number of times in the main code, I feel we should have a package constant somewhere.

Reply:
I was being lazy :-)